xtask/src/main.rs

```diff
@@ -0,0 +1,605 @@
+            panic!("Error installing targets: {}", added_targets.join(" "));
+        }
+    }
+
+    Ok(added_targets)
+}
+
+fn uninstall_targets(targets: Vec<String>) {
+    if !targets.is_empty() {
+        println!("⏳ uninstalling targets");
+
+        let mut cmd_and_args: Vec<&str> = vec!["rustup", "target", "remove"];
+        cmd_and_args.extend(targets.iter().map(|s| s.as_str()));
+
+        // only print uninstall errors so the user can fix those manually if needed
+        match run_command::<&str>(&cmd_and_args, None, &[]) {
+            Ok(_) => {}
+            Err(e) => {
+                eprintln!("Error uninstalling targets {}: {}", targets.join(" "), e);
+            }
+        }
+    }
+}
+
+fn test_book(errors: &mut Vec<String>) {
+    println!("🧪 book");
+    do_test(
+        || run_command::<&str>(&["cargo", "clean"], None, &[]),
+        "book",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command::<&str>(
+                &["cargo", "build", "--features", "unstable-test"],
+                None,
+                &[],
+            )
+        },
+        "book",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "mdbook",
+                    "test",
+                    "-L",
+                    "../target/debug",
+                    "-L",
+                    "../target/debug/deps",
+                ],
+                Some("book"),
+                &[],
+            )
+        },
+        "book",
+        errors,
+    );
+}
+
+fn test_lint(errors: &mut Vec<String>) {
+    println!("🧪 lint");
+    do_test(
+        || run_command::<&str>(&["cargo", "clean"], None, &[]),
+        "lint",
+        errors,
+    );
+    do_test(
+        || run_command::<&str>(&["cargo", "fmt", "--all", "--", "--check"], None, &[]),
+        "lint",
+        errors,
+    );
+
+    do_test(
+        || run_command::<&str>(&["cargo", "clippy", "--workspace"], None, &[]),
+        "lint",
+        errors,
+    );
+}
+
+fn test_host(deny_warnings: bool, errors: &mut Vec<String>) {
+    println!("🧪 host");
+
+    let env = if deny_warnings {
+        vec![("RUSTFLAGS", "--deny warnings")]
+    } else {
+        vec![]
+    };
+
+    do_test(
+        || run_command::<&str>(&["cargo", "check", "--workspace"], None, &env),
+        "host",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command::<&str>(
+                &["cargo", "check", "--all", "--features", "unstable-test"],
+                None,
+                &env,
+            )
+        },
+        "host",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command::<&str>(
+                &["cargo", "check", "--all", "--features", "alloc"],
+                None,
+                &env,
+            )
+        },
+        "host",
+        errors,
+    );
```

Comment:
This pattern seems to take up quite some space.. Can we make it a bit less spacious?

Maybe:
```rust
    let cmd_and_args = &["cargo", "check", "--all", "--features", "alloc"];
    do_test(
        || run_command::<&str>(cmd_and_args, None, &env),
        "host",
        errors,
    );
```

Reply:
`rustfmt` won't

Reply:
hm. I like consistency, which means here to introduce a temp variable for every `do_test` call. That's also verbose in its own way (and harder to type when adding a test).

Reply:
I agree that the separate variable is also bad. But there has to be a better way! :P

Reply:
... `#[rustfmt::skip]`? 😬 (I'm only half joking)

Reply:
Naah 🤣

Reply:
I mean, you can annotate on block level, so ... it's not a _terrible_ idea.

Reply:
This actually isnt a blocking issue. Lets just keep it like it is, and fix it if we find a good solution and not the hacka we were thinking off.
(btw: increasing the rustfmt max_width to 115 flattens many of the tests :D)

Reply:
I actually like this approach a lot. Have committed a corresponding `rustfmt.toml` and the reformatted code

Reply:
people with small screens will hate us, but yes, seems like the best workaround for now 😆

Reply:
play small screen games, win small font prizes